File: 백준/Silver/9012. 괄호/괄호.swift
let T = Int(readLine()!)!
for _ in 0..<T {
    let input = Array(readLine()!)
    var count = 0
    
    for char in input {
        count += char == "(" ? 1 : -1
        if count < 0 {
            break
        }
    }
    
    print(count == 0 ? "YES" : "NO")
}

File: 백준/Silver/9012. 괄호/괄호.py
t = int(input())
result = ""
for _ in range(t):
    vps = input().strip()
    count = 0
    for char in vps:
        if char == '(':
            count += 1
        else:
            count -= 1
        if count < 0:
            break
    if count == 0:
        result += "YES\n"
    else:
        result += "NO\n"
print(result)